### ISSUE TITLE
Time to compare again? (webcompents/custom-elements)

### ISSUE THREAD

**Issue by wbern** [WebReflection/document-register-element#181]
Thank you for all your efforts. This package is a cornerstone for many of us that advocate a modern web, but are held down by the remaining old browsers.

I was wondering, after 4 years, what would be the considered status on issue https://github.com/WebReflection/document-register-element/issues/81, namely webcomponentsjs/custom-element polyfill and DRE?

My org are now in the process of comparing and mixing polyfills. We got ShadyDOM to work ok, and we need it for slots. We don't need ShadyCSS or anything else, really, just the slots support.

I am now thinking of whether we'd want to use DRE (which has been great to us), and how compatible it would be in theory with ShadyDOM.

Apologies if this question feels out of line, we're doing research now, but any comments on your end (the mighty 1 maintainer!) would be sincerely welcome.

In the mean time, be well and careful.

**Comment by WebReflection** [comment]
So, this project has been about [Custom Elements](https://html.spec.whatwg.org/multipage/custom-elements.html) **only** since about ever, and its successor is [@ungap/custom-elements-builtin](https://github.com/ungap/custom-elements-builtin#readme).

[Web Components](https://www.w3.org/wiki/WebComponents/), as defined by standards, have (used to be 4 but ...) *3* sub proposals: _Shadow DOM_, _Custom Elements_, and the AFAIK dead _HTML Imports_ idea.

Accordingly, you'll never find _Shadow DOM_ in here for at least two reasons:
  * _Custom Elements_, and builtin extends, don't need ShadowDOM, it's a specification a part
  * _ShadowDOM_ **cannot be polyfilled**, so that the closest attempt I've made is [attachShadow](https://github.com/WebReflection/attachshadow#readme), which provides an extremely minimal attempt at simulating _ShadowDOM_.

In few words, _ShadowDOM_ is a specification a part, it cannot be reliably polyfilled, because old browsers don't have its capabilities, and every polyfill for _ShadowDOM_ is inherently broken, but evangelists/advocates wont tell this story to anyone.

*TL;DR* no, shadow DOM has no place in here, because it's a polyfill a part, and it's a broken promise itself, as polyfill.

Apologies for not answering the way you hoped for, but that's just the reality.

**Comment by WebReflection** [comment]
by the way, if you're looking for a helper that doesn't need ShadowDOM and yet let you configure in a better way CSS for any Custom Element, check [µce](https://github.com/WebReflection/uce#readme) out, as it covers it all, without ShadowDOM in between (and yet, it's compatible with ShadowDOM, so you have a future proof helper right there)
